posydon/binary_evol/SN/step_SN.py

```diff
@@ -927,6 +930,55 @@ def PISN_prescription(self, star):
                         m_PISN = m_star
                     else:
                         m_PISN = m_He_core
+            
+            elif self.PISN == 'Hendriks+23':
+                # Hendriks et al. 2023 PISN prescription
+                # Shifting PPI and PISN gap
+                # works by removing delta_M_PPI from the star
+                # and then applying any remnant mass prescription
+                
+                delta_M_CO_shift = self.PISN_CO_shift if self.PISN_CO_shift is not None else 0.0
+                delta_M_PPI_extra_ML = self.PPI_extra_mass_loss if self.PPI_extra_mass_loss is not None else 0.0
+                if ((m_CO_core >= 38 + delta_M_CO_shift) 
+                    and m_CO_core <= 114 + delta_M_CO_shift):
+
+                    delta_M_PPI = (
+                        (0.0006 * np.log10(star.metallicity * const.Zsun) + 0.0054)
+                        * (m_CO_core - delta_M_CO_shift - 34.8)**3
+                        - 0.0013 * (m_CO_core - delta_M_CO_shift - 34.8)**2
+                        + delta_M_PPI_extra_ML
+                    )
```

Comment:
- May add a comment here, which equation this is in the paper
- May consider to define a variable for `m_CO_core - delta_M_CO_shift`, which looks to be often used (incl. the condition)
- Shell we limit the metallicity? Because currently for `Z -> 0` we get `delta_M_PPI -> - infinity`. In the paper it looks like they only considered log10(Z)>=-4.